src/openrct2/ride/Station.cpp

```diff
@@ -284,23 +284,31 @@ static void ride_race_init_vehicle_speeds(Ride* ride)
         {
             Peep* peep = &get_sprite(vehicle->peep[0])->peep;
 
-            switch (peep_get_easteregg_name_id(peep))
+            // Easter egg names should only work on guests
+            // Check to make sure this peep is a guest before applying easter egg logic
+
+            Guest* guest = peep->AsGuest();
+
+            if (guest)
```

Comment:
Always explicitly check for nullptr:
```suggestion
            if (guest != nullptr)
```